### ISSUE TITLE
generated JavaScript for polymer elements dosn't call xxxChanged for published attributes

### ISSUE THREAD

**Issue by DartBot** [dart-lang/polymer-dart#369]
<a href="https://github.com/zoechi"><img src="https://avatars.githubusercontent.com/u/405837?v=3" align="left" width="96" height="96"hspace="10"></img></a> **Issue by [zoechi](https://github.com/zoechi)**
_Originally opened as dart-lang/sdk#15080_

----

**What steps will reproduce the problem?**
**1.**
We run into this problem 
https://code.google.com/p/dart/issues/detail?id=15038
when porting 
- https://github.com/ErikGrimes/polymer_elements
- https://github.com/ErikGrimes/polymer_ui_elements
which have a demo page for each polymer element in the 'example' directory.
So we changed the package layout as suggested:
We created a new package and symlinked the demo pages from example
to the new packages web directory
removed the transformers settings in polymer_elements/polymer_ui_elements
and added them to the pubspec.yaml of the new package.
**2.**
pub build works fine
But in all JS demo pages the xxxChanged methods for published attributes never get called (except when called from other methods directly)
Methods like enteredView or ready are called though.

**3.**
No warning or error is shown (build/execution)

**What is the expected output? What do you see instead?**
Should work in JS as in Dart

**What version of the product are you using? On what operating system?**
Dart VM version: 1.0.0.3_r30187 (Tue Nov 12 00:09:52 2013) on &quot;linux_x64&quot;

**Please provide any additional information below.**
Before we changed the package structure as described above some examples worked fine in JS (those that were not affected by the issue linked above)
I also tried copying the files to the new package instead of symlinking but the problem persisted.

**Comment by DartBot** [comment]
<a href="https://github.com/sethladd"><img src="https://avatars.githubusercontent.com/u/5479?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sethladd](https://github.com/sethladd)**

----

*Added [Area-Polymer](../labels/Area-Polymer), [Triaged](../labels/Triaged) labels.*

**Comment by DartBot** [comment]
<a href="https://github.com/zoechi"><img src="https://avatars.githubusercontent.com/u/405837?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [zoechi](https://github.com/zoechi)**

----

When I add 
transformers:
- polymer:
&nbsp;&nbsp;&nbsp;&nbsp;entry_points:
(without any actual entry_points defined) to pubspec.yaml of the component package
the generated JS works even when the package isn't supposed to have a polymer transformer specified.

Dart VM version: 1.0.1.3_r30657 (Mon Nov 25 18:29:39 2013) on "linux_x64"

**Comment by DartBot** [comment]
<a href="https://github.com/sigmundch"><img src="https://avatars.githubusercontent.com/u/2049220?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sigmundch](https://github.com/sigmundch)**

----

*Added [Library-PolymerBuild](../labels/Library-PolymerBuild) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/zoechi"><img src="https://avatars.githubusercontent.com/u/405837?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [zoechi](https://github.com/zoechi)**

----

I think this is related to 18489

**Comment by DartBot** [comment]
<a href="https://github.com/sigmundch"><img src="https://avatars.githubusercontent.com/u/2049220?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sigmundch](https://github.com/sigmundch)**

----

*Added this to the [1.5](../milestones/1.5) milestone.*

**Comment by DartBot** [comment]
<a href="https://github.com/jmesserly"><img src="https://avatars.githubusercontent.com/u/1081711?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [jmesserly](https://github.com/jmesserly)**

----

*Added [Pkg-Polymer](../labels/Pkg-Polymer) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/jmesserly"><img src="https://avatars.githubusercontent.com/u/1081711?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [jmesserly](https://github.com/jmesserly)**

----

*Removed [Pkg-Polymer](../labels/Pkg-Polymer) label.*
*Added [Pkg-PolymerBuild](../labels/Pkg-PolymerBuild) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/jmesserly"><img src="https://avatars.githubusercontent.com/u/1081711?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [jmesserly](https://github.com/jmesserly)**

----

*Removed [Library-PolymerBuild](../labels/Library-PolymerBuild) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/jmesserly"><img src="https://avatars.githubusercontent.com/u/1081711?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [jmesserly](https://github.com/jmesserly)**

----

*Removed [Area-Polymer](../labels/Area-Polymer) label.*
*Added [Area-Pkg](../labels/Area-Pkg) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/kasperl"><img src="https://avatars.githubusercontent.com/u/2156198?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [kasperl](https://github.com/kasperl)**

----

Postponing to 1.6. Feel free to change this back (and give it a proper priority).
______
*Removed this from the [1.5](../milestones/1.5) milestone.*
*Added this to the [1.6](../milestones/1.6) milestone.*

**Comment by DartBot** [comment]
<a href="https://github.com/zoechi"><img src="https://avatars.githubusercontent.com/u/405837?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [zoechi](https://github.com/zoechi)**

----

I still had this a while ago (when I added the comment at May 3) but this seems not to be an issue on all projects. Maybe this is only when the project has Polymer elements in lib and also uses elements imported from another package.

I didn't have this in other projects and this is the only notable distinction.

**Comment by DartBot** [comment]
<a href="https://github.com/sigmundch"><img src="https://avatars.githubusercontent.com/u/2049220?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sigmundch](https://github.com/sigmundch)**

----

Thanks for the details!

I forgot to mention earlier, but I agree with you in #­4 that this will likely be fixed by issue dart-lang/sdk#9.
______
*Marked this as being blocked by dart-lang/code_transformers#17.*

**Comment by DartBot** [comment]
<a href="https://github.com/sigmundch"><img src="https://avatars.githubusercontent.com/u/2049220?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sigmundch](https://github.com/sigmundch)**

----

*Removed [Priority-Unassigned](../labels/Priority-Unassigned) label.*
*Added [Priority-Medium](../labels/Priority-Medium) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/sigmundch"><img src="https://avatars.githubusercontent.com/u/2049220?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sigmundch](https://github.com/sigmundch)**

----

*Removed this from the [1.6](../milestones/1.6) milestone.*
*Added [Polymer-P-1](../labels/Polymer-P-1) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/sigmundch"><img src="https://avatars.githubusercontent.com/u/2049220?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sigmundch](https://github.com/sigmundch)**

----

*Removed [Polymer-P-1](../labels/Polymer-P-1) label.*
*Added [Polymer-Milestone-Next](../labels/Polymer-Milestone-Next) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/sigmundch"><img src="https://avatars.githubusercontent.com/u/2049220?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sigmundch](https://github.com/sigmundch)**

----

*Added [PolymerMilestone-Next](../labels/PolymerMilestone-Next) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/sigmundch"><img src="https://avatars.githubusercontent.com/u/2049220?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sigmundch](https://github.com/sigmundch)**

----

*Removed [Polymer-Milestone-Next](../labels/Polymer-Milestone-Next) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/sigmundch"><img src="https://avatars.githubusercontent.com/u/2049220?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sigmundch](https://github.com/sigmundch)**

----

merging with issue dart-lang/sdk#20267 to track things together.
______
*Added [Duplicate](../labels/Duplicate) label.*
*Marked as being merged into dart-lang/sdk#20267.*
